### PR TITLE
Move registration into its own module.

### PR DESCRIPTION
**Pull request by ttung**
Changed maxproject into a specific algorithm of registration to enable multiple algorithms.

### CODE REVIEW

File: starfish/registration/__init__.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    @classmethod
+    def register(cls, args):
+        if args.registration_algorithm is None:
+            cls.register_group.print_help()
+            cls.register_group.exit(status=2)
+
+        args.registration_algorithm(args)
+
+    @classmethod
+    def maxproject(cls, args):
```
Comment (dganguli) [inline starfish/registration/__init__.py]: rename maxproject -> fourier_shift.

File: tests/test_iss_data.py [hunk 2]
```diff
@@ -34,6 +34,7 @@ class TestWithIssData(unittest.TestCase):
             "starfish", "register",
             lambda tempdir, *args, **kwargs: os.path.join(tempdir, "formatted", "org.json"),
             lambda tempdir, *args, **kwargs: os.path.join(tempdir, "registered"),
+            "maxproject",
```
Comment (dganguli) [inline tests/test_iss_data.py]: "fourier_shift",

File: starfish/registration/__init__.py (hunk 1 above)
Comment (ttung) [reply starfish/registration/__init__.py]: done.

File: tests/test_iss_data.py (hunk 2 above)
Comment (ttung) [reply tests/test_iss_data.py]: done.

**Pull request merged**